src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs

```diff
@@ -45,8 +50,10 @@ public class PackageDetailControlModel : DetailControlModel
             {
                 return;
             }
+            PreviousSelectedVersion = null;
             InstalledVersion = searchResultPackage.InstalledVersion;
-            SelectedVersion.IsCurrentInstalled = InstalledVersion == SelectedVersion.Version;
+            InstalledVersionRange = searchResultPackage.AllowedVersions;
+            SelectedVersion.IsCurrentInstalled = InstalledVersion == SelectedVersion.Version && InstalledVersionRange == SelectedVersion.Range;
```

Comment:
Bug risk: are these Version Range properties always `null == null` or set to real values as expected?